src/extensions/try-convert/MSBuild.Abstractions/ProjectItemComparer.cs

```diff
@@ -38,20 +40,43 @@ public bool Equals([AllowNull] IProjectItem x, [AllowNull] IProjectItem y)
                 return false;
             }
 
-            // If y has all the metadata that x has then we declare them as equal. This is because
-            // the sdk can add new metadata but there's not reason to remove them during conversion.
-            var metadataEqual = _compareMetadata ?
-                                 x.DirectMetadata.All(xmd => y.DirectMetadata.Any(
-                                     ymd => xmd.Name.Equals(ymd.Name, System.StringComparison.OrdinalIgnoreCase) &&
-                                            xmd.EvaluatedValue.Equals(ymd.EvaluatedValue, System.StringComparison.OrdinalIgnoreCase)))
-                                 : true;
+            if (x.ItemType != y.ItemType)
+            {
+                return false;
+            }
+
+            var xPath = PathHelpers.GetIncludePath(x.EvaluatedInclude);
+            var yPath = PathHelpers.GetIncludePath(y.EvaluatedInclude);
+
+            if (!xPath.Equals(yPath, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            if (_compareMetadata)
+            {
+                // If y has all the metadata that x has then we declare them as equal. This is because
+                // the sdk can add new metadata but there's not reason to remove them during conversion.
+                var metadataEqual = x.DirectMetadata.All(xmd => y.DirectMetadata.Any(
+                                         ymd => xmd.Name.Equals(ymd.Name, StringComparison.OrdinalIgnoreCase) &&
+                                                xmd.EvaluatedValue.Equals(ymd.EvaluatedValue, StringComparison.OrdinalIgnoreCase)));
 
-            return x.ItemType == y.ItemType && x.EvaluatedInclude.Equals(y.EvaluatedInclude, System.StringComparison.OrdinalIgnoreCase) && metadataEqual;
+                if (!metadataEqual)
+                {
+                    return false;
+                }
+            }
+
+            return true;
         }
 
         public int GetHashCode(IProjectItem obj)
         {
-            return (obj.EvaluatedInclude.ToLowerInvariant() + obj.ItemType).GetHashCode();
+            obj = obj ?? throw new ArgumentNullException(nameof(obj));
+
+            var path = PathHelpers.GetIncludePath(obj.EvaluatedInclude);
+
+            return (obj.ItemType + ":" + path).GetHashCode(StringComparison.Ordinal);
         }
```

Comment:
Inconsistent comparison with line 51. On the other hand this seems rather complicated for just a hashcode anyway. I'm used to seeing just things like:

``` C#
return (obj.ItemTypeGetHashCode() ^ path.GetHashCode();
```